### ISSUE TITLE
skflow save sets absolute paths for checkpoint that break on production

### ISSUE THREAD

**Issue by nemo** [tensorflow/tensorflow#1852]
### Environment info
Running on latest [nightly build](http://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_CONTAINER_TYPE=CPU,TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-0.7.1-cp27-none-linux_x86_64.whl).

### Steps to reproduce
1. Save an estimator to a folder (let's call it classifier.model)
2. Attempt to restore on another machine in different machine

This breaks since the checkpoint file is pointing to the place in the new machine.

### What have you tried?
1. Manually changing the path in the checkpoint file fixes the issue

**Comment by ilblackdragon** [comment]
@sherrym skflow currently uses Saver to save checkpoints - is there anyway to disable saving global path into `checkpoint` file and save local path instread?

**Comment by ilblackdragon** [comment]
@nemo Actually looking [at this](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/saver.py#L558) code, it seems like if you just specify relative path when you are saving (e.g. `est.save('my_model/')`) - it will save with relative path and should work. Can you tell if this fixes your problem?

**Comment by nemo** [comment]
@ilblackdragon just tried it. Still seeing absolute paths in there - we had `./my_model` before and I tried `my_model` and `my_model/`, both of which still created a checkpoint file with absolute paths.

**Comment by sherrym** [comment]
@nemo , could you please cut and paste your code showing how you set the save_path? I have unit tests demonstrating that relative paths work correctly.

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/saver_test.py#L659

**Comment by nemo** [comment]
Here it is:
```
classifier = learn.TensorFlowEstimator(model_fn=tanh_dnn, n_classes=len(categories) + 1,
    steps=len(documents) * 10, learning_rate=0.2, batch_size=100)

classifier.fit(X_train, y_train)
classifier.save('categories.model/')
```

Since it was renamed from `skflow` to `learn` - I haven't tested the different path types other than the one you see there, which still causes the checkpoint file to have a full path.

**Comment by sherrym** [comment]
@ilblackdragon, Saver.save() takes a save_path as an argument. You just need to change TensorFlowEstimator's save() in estimators/base.py to pass in save_path.

**Comment by michaelisard** [comment]
Closing due to lack of activity. Please reopen if there is still a problem.
